src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends { [key: string]: string }>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  return items.reduce((acc, item) => {
+    const prev = acc[item[key]] ?? [];
+
+    return {
+      ...acc,
+      [item[key]]: [...prev, item],
+    };
+  }, {} as GroupsMap<T>);
```

Comment:
Inside the reduce callback, the spread operator is used to clone the accumulator object and the array of grouped items. While this is not incorrect, it can lead to performance issues if the 'items' array is large because it creates a new array and a new object on each iteration. Instead, you can directly modify the accumulator object since it's scoped to the reduce function and will not affect any external state. This way, you avoid the additional overhead of creating new arrays and objects on each iteration.